modules/pico_mdns.c

```diff
@@ -762,7 +767,13 @@ pico_mdns_generate_new_records( pico_mdns_rtree *conflict_records,
                                              PICO_MDNS_RECORD_CURRENTLY_PROBING));
 
             /* Add the record to the new tree */
-            pico_tree_insert(&new_records, new_record);
+            if(pico_tree_insert(&new_records, new_record) ){
+            	mdns_dbg("Could not add new non-conflicting record to the tree!\n");
+				if(pico_err != PICO_ERR_ENOMEM){
+					pico_err = PICO_ERR_EINVAL;
+				}
+				return new_records;
+			}
```

Comment:
Doesn't the `new_record` need to be freed too, somehow? Maybe @jelledevleeschouwer can shed some light on this?